libs/ui/src/components/MediaItem/MediaItem.vue

```diff
@@ -92,6 +91,35 @@ const props = withDefaults(
   },
 )
 
+const mediaItem = ref<HTMLDivElement>()
+let modelComponent: Component | null = null
+let isModelVisible = ref(false)
+let observer: IntersectionObserver
+
+onMounted(() => {
+  observer = new IntersectionObserver((entries) => {
+    if (entries.some((entry) => entry.isIntersecting)) {
+      if (mimeType.value === 'model/gltf-binary') {
+        isModelVisible.value = true
+        modelComponent = defineAsyncComponent(
+          () => import('./type/ModelMedia.vue'),
```

Comment:
need to make sure this one, sometimes the components were not loaded

<img width="600" alt="Screenshot 2023-11-13 at 10 40 14 PM" src="https://github.com/kodadot/nft-gallery/assets/734428/c264b693-1519-4688-ab0e-7167ace459c2">

https://deploy-preview-8030--koda-canary.netlify.app/ksm/explore/items?listed=true

Reply:
Done